src/services/events/utils/types.ts

```diff
@@ -0,0 +1,101 @@
+import { CartDto, Product, User } from '../../../interfaces'
+
+export type EventCallbackFunction<Payload = undefined> = (payload: Payload) => void
+
+export type GetTypeFromInterface<T> = T extends EventCallbackFunction<infer Payload>
+  ? Payload
+  : undefined
+
+export enum EventType {
+  addToCart = 'addToCart',
+  addToWishlist = 'addToWishList',
+  completeRegistration = 'completeRegistration',
+  contact = 'contact',
+  customizeProduct = 'customizeProduct',
+  donate = 'donate',
+  findLocation = 'findLocation',
+  initiateCheckout = 'initiateCheckout',
+  lead = 'lead',
+  onPurchase = 'onPurchase',
+  removeFromCart = 'removeFromCart',
+  schedule = 'schedule',
+  search = 'search',
+  signUp = 'signUp',
+  viewContent = 'viewContent',
+}
+
+export interface HeseyaEvents {
```

Comment:
Może łatwiej będzie wyciągać typy payloada, jeśli zamiast takiego złożonego typu będzie typ będący mapą `HeseyaEventType` do jego `Payload`

Wtedy zarówno w przypadku `emit` i `on` będzie można brać `EventPayloadMap[Key]` i może będzie to czytelniejsze niż teraz